src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c

```diff
@@ -28,24 +29,141 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
     (ActorFunc)ObjTokeiTobira_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80ABD750[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 300, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 300, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+Vec3f D_80ABD760 = { 0.0f, 0.0f, 80.0f };
+s16 D_80ABD76C[] = { -0x4000, 0x4000 };
+CollisionHeader* D_80ABD770[] = { &object_tokei_tobira_Colheader_0012B0, &object_tokei_tobira_Colheader_001590 };
+f32 D_80ABD778[] = { 1.0f, -1.0f };
+Gfx* D_80ABD780[] = { object_tokei_tobira_DL_001108, object_tokei_tobira_DL_0013E8 };
 
-extern InitChainEntry D_80ABD750[];
+void ObjTokeiTobira_Init(Actor* thisx, PlayState* play) {
+    ObjTokeiTobira* this = THIS;
+    s32 pad;
+    s32 type = OBJTOKEITOBIRA_GET_TYPE(&this->dyna.actor);
+    Vec3f pos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Init.s")
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, DYNA_TRANSFORM_POS | DYNA_TRANSFORM_ROT_Y);
+    DynaPolyActor_LoadMesh(play, &this->dyna, D_80ABD770[type]);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Destroy.s")
+    if (type == OBJTOKEITOBIRA_TYPE_0) {
+        Actor_SpawnAsChild(&play->actorCtx, &this->dyna.actor, play, ACTOR_OBJ_TOKEI_TOBIRA,
+                           this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z,
+                           this->dyna.actor.shape.rot.x, this->dyna.actor.shape.rot.y, this->dyna.actor.shape.rot.z,
+                           OBJTOKEITOBIRA_PARAMS(OBJTOKEITOBIRA_TYPE_1));
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/func_80ABD3B0.s")
+    Matrix_RotateYS(D_80ABD76C[type] + this->dyna.actor.shape.rot.y, MTXMODE_NEW);
+    Matrix_MultVec3f(&D_80ABD760, &pos);
+    this->dyna.actor.world.pos.x += pos.x;
+    this->dyna.actor.world.pos.y += pos.y;
+    this->dyna.actor.world.pos.z += pos.z;
```

Comment:
```suggestion
    this->dyna.actor.world.pos.x += posOffset.x;
    this->dyna.actor.world.pos.y += posOffset.y;
    this->dyna.actor.world.pos.z += posOffset.z;
```
Position isn't really correct here. It's more an offset from the base position